packages-node/nodejs-helpers/test-node/tasks/bypass-export-map.test.js

```diff
@@ -0,0 +1,328 @@
+import {
+  prettify,
+  asyncConcurrentForEach,
+  byStringAscendingSort,
+  getExportSpecifiersByFile,
+  transformCode,
+  bypassExportMap,
+} from '@lion/nodejs-helpers';
+
+import {
+  ERROR_CAN_NOT_OVERWRITE_EXISTING_FILE,
+  ERROR_CAN_NOT_RESOLVE_SOURCE,
+  // ERROR_ADJUSTED_SOURCE_IS_INVALID,
+  ERROR_CAN_NOT_ACCESS_PACKAGE_DIR,
+} from '../../src/tasks/bypass-export-map.js';
+
+// Register chai-as-promised plugin for async/await support.
+chai.use(chaiAsPromised);
+const { expect } = chai;
+
+/**
+ * Get sorted js files for the given directory
+ *
+ * @param {string} searchPattern The globby search pattern
+ * @returns {Promise<string[]>}
+ */
+const getSortedJsFileNamesInDir = async searchPattern => {
+  const files = await globby(searchPattern);
+  return files.map(file => path.basename(file)).sort(byStringAscendingSort);
+};
+
+/**
+ *
+ * @param {string[]} fsPaths FileSystem paths to remove with `rm -rf`
+ * @returns {Promise<void[]>}
+ */
+const rmRecursiveForce = async fsPaths => {
+  // @ts-ignore
+  const deleteFsPath = fsPath => rm(fsPath, { recursive: true, force: true });
+  return asyncConcurrentForEach(fsPaths, deleteFsPath);
+};
+
+/**
+ * Get export specifiers for all the js files in a directory given with `dirPath`
+ * @param {string} dirPath Directory path
+ * @returns {Promise<string[]>}
+ */
+const getExportSpecifiersByDir = async dirPath => {
+  const fileNames = await getSortedJsFileNamesInDir(path.join(dirPath, '*.js'));
+  // @ts-ignore
+  const exports = await asyncConcurrentForEach(fileNames, async fileName => {
+    const filePath = path.resolve(dirPath, fileName);
+    return getExportSpecifiersByFile(filePath);
+  });
+  return exports.flat().sort(byStringAscendingSort);
+};
+
+/**
+ * Get the paths returned by require.resolve for each
+ * import or export declaration found in the given `code`
+ * @param {string} code
+ * @param {{paths: string[]}} options
+ * @returns {Promise<string[]>}
+ */
+const getRequireResolvePaths = async (code, { paths = [] }) => {
+  /**
+   * @type {string[]}
+   */
+  const resolvedPaths = [];
+  const visitor = {
+    // @ts-ignore
+    enter({ node }) {
+      const isImportExportNode = isImportDeclaration(node) || isExportDeclaration(node);
+      if (!isImportExportNode) {
+        return;
+      }
+      // @ts-ignore
+      const { source } = node;
+      // example: `import sth from source.value;`
+      const initialSource = source?.value;
+      const isRelativeSource = initialSource?.startsWith('.');
+      if (!isRelativeSource) {
+        return;
+      }
+      const require = createRequire(import.meta.url);
+      const resolvedPath = require.resolve(initialSource, { paths });
+      resolvedPaths.push(resolvedPath);
+    },
+  };
+  transformCode(code, visitor);
+  return resolvedPaths;
+};
+
+describe('bypassExportMap simple export map case', async () => {
+  const packageDir = 'test-node/fixtures/simple-export-map';
```

Comment:
Did you consider using mock-fs (for readability and perf)?

Then we can do:

```js
mockFs({
  '/simple-export-map/components/icon/xyz.js': "export x = () => 'bla';"
})

```

Reply:
Yes I did consider mocking filesystem, however it is another layer of abstraction on top of the reality, which will make it less readable imo, and more of a complex solution. Of course, as you suggested we could still consider it, say for performance, and that would certainly be the case, if we were mocking timers and so on, yet performance wise the whole tests that access the filesystem are taking ~0.247 seconds to be precise.

To get the ~0.247, we can

First run the tests without the tests which uses the filesystem

`rm -rf packages-node/nodejs-helpers/test-node/tasks/ && time npm run test -w packages-node/nodejs-helpers/`
 
real	0m1.323s

And then, run it with them: 
`g checkout packages-node/nodejs-helpers/test-node/tasks/ && time npm run  test -w packages-node/nodejs-helpers/ `

real	0m1.570s

Then the difference is ~0.247,, which is insignificant.